lib/syslog/syslog.go

```diff
@@ -99,17 +97,29 @@ func Init() {
 		panic(fmt.Errorf("Syslog is configured but configuration file missing"))
 	}
 
-	// Initializes the buffered channel and the syslog writer 
+	// Initializes the buffered channel and the syslog writer
 	logChan = make(chan SyslogLogContent, sysCfg.QueueConfig.Capacity)
-	syslogW := &syslogWriter{framer: defaultFramer, formatter: defaultFormatter}
+
+	var syslogW *syslogWriter
+	if sysCfg.SyslogConfig.Protocol == "tcp+tls" {
+		var syslogtlsConfig *tls.Config
+		syslogtlsConfig, err := readTlsConfig(sysCfg.SyslogConfig.Tls.CertFile, sysCfg.SyslogConfig.Tls.KeyFile, sysCfg.SyslogConfig.Tls.Ca, sysCfg.SyslogConfig.Tls.ServerName, sysCfg.SyslogConfig.Tls.InsecureSkipVerify)
+		if err != nil {
+			fmt.Errorf("Failed reading the tls certificates %w", err)
+		}
+		syslogW = &syslogWriter{framer: defaultFramer, formatter: defaultFormatter, tlsConfig: syslogtlsConfig}
```

Comment:
Construction of syslogWriter can be done via method chaining so that line 106 and 110 can be common. wdyt?
`syslogW = syslogWriter.New().WithFramer(framer).WithFormatter(formatter).WithTlsConfig(tlsconfig)`